executor/builder.go

```diff
@@ -117,6 +117,25 @@ func (b *executorBuilder) buildFilter(src Executor, conditions []ast.ExprNode) E
 }
 
 func (b *executorBuilder) buildTableScan(v *plan.TableScan) Executor {
+	txn, err := b.ctx.GetTxn(false)
+	if err != nil {
+		b.err = err
+		return nil
+	}
+	xapi := txn.GetXAPI()
+	if xapi != nil {
+		e := &XSelectTableExec{
+			ctx:       b.ctx,
+			tablePlan: v,
+		}
+		where := conditionsToPBExpression(v.FilterConditions)
+		if xapi.SupportExpression(where) {
```

Comment:
Even if expression is not supported, we can still use xapi, we just filter the result in tidb-server side.